ghost/core/core/server/services/settings/SettingsBREADService.js

```diff
@@ -194,6 +197,12 @@ class SettingsBREADService {
         }
 
         if (stripeConnectData) {
+            if (this.limitsService.isDisabled('limitStripeConnect')) {
+                throw new NoPermissionError({
+                    message: tpl(messages.stripeConnectDisabled)
+                });
+            }
```

Comment:
I should've mentioned that earlier, but the limit service provides you with the correct error message and type. There's no clean method to get it, particularly while staying in sync and not switching to async code. But all you'd need to change is using this line instead of throwing the error:

`await limitService.errorIfWouldGoOverLimit('limitStripeConnect');`

This will throw the correct `HostLimitError` with the error message defined on the limit in our DB.

Reply:
Ooh, I'll have a quick go at that